src/connectionbenchmark/uploaddatagenerator.h

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef UPLOADDATAGENERATOR_H
+#define UPLOADDATAGENERATOR_H
+
+#include <QIODevice>
+
+class UploadDataGenerator : public QIODevice {
+  Q_OBJECT
+  Q_DISABLE_COPY_MOVE(UploadDataGenerator)
+
+ public:
+  explicit UploadDataGenerator(const qint64& totalSize);
+  virtual ~UploadDataGenerator();
+
+  qint64 size() const override {
+    return m_totalSize;
```

Comment:
Should this return the total size of what left?

Reply:
I’m not entirely sure — I think [size()](https://doc.qt.io/qt-6/qiodevice.html#size) is supposed to return the total size that the device is “supposed” to contain?